modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/model/impl/ComponentManagerImpl.java

```diff
@@ -387,11 +387,16 @@ public void removeListener(ComponentManager.Listener listener) {
     public ComponentInstance getComponentProvidingService(Class<?> serviceClass) {
         RegistrationInfo ri = services.get(serviceClass.getName());
         if (ri == null) {
+            log.debug("The component exposing the service: {} doesn't exist", serviceClass);
             return null;
         }
         ComponentInstance ci = ri.getComponent();
         if (ci == null) {
-            log.debug("The component exposing the service {} is not resolved or not started", serviceClass);
+            log.debug("The component exposing the service: {} is not resolved", serviceClass);
+        }
+        if (Boolean.getBoolean("nuxeo.start.strict") && ri.getState() == RegistrationInfo.START_FAILURE) {
```

Comment:
Maybe `ri.getState() == RegistrationInfo.START_FAILURE` first because strict mode will always be true and getBoolean is slower